### ISSUE TITLE
Is it possible hidding files?

### ISSUE THREAD

**Issue by geaplanet** [gobolinux/GoboHide#5]
I'm playing with gobohide and it is really cool. I missed a tool like this that truly hidden files on the system without "dot" tricks or filemanager dependency. I was wondering why can only folders and symlinks to be hidden? why not files? Is it a filesystem limitation?

**Comment by lucasvr** [comment]
On the kernel side, all entries to be hidden are put in a simple linked list. Since we typically have very few entries (currently 16), iterating over that list on a call to readdir() is not that expensive. However, if we were to allow regular files to be hidden as well then the size of that list would likely grow to a much greater extent, as users would potentially want to hide large sets of files, possibly impacting readdir() times. Hence, we have opted to support directories and symlinks alone.

https://github.com/gobolinux/GoboHide/blob/master/src/gobohide.c#L190 is where that restriction needs to be removed if so you wish. The kernel patch, as far as I remember, does not enforce the same restriction as the userspace tool.

**Comment by geaplanet** [comment]
Wow! Thank you very much. I did what you told me, I removed the restriction and I did some tests. I don't think I will ever need to hide thousands of files, and if I had to, I'd better hide the folder. But I have hidden thousands of files (3,000) and it doesn't take a second. Then to get them back I did a list and iterate it in bash, it is fast too (2 or 3 seconds). 

Offtopic:
I love the idea of Gobolinux project, but I need to try it for a longer time to doing things to the gobo style. I have been using only Debian for too long : -/ Right now I am using it in virtualbox, the only thing that I have noticed is that it takes a long time to boot and, taking into account that at the end of the booting it leaves you in a terminal, it is very strange. I will continue investigating and what I find I'll put in another threads.
